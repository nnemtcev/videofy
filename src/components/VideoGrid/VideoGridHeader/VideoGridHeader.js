import React from "react";
import "./VideoGridHeader.scss";

// This component is for rendering a customizable header
// for each VideoGrid component

export function VideoGridHeader(props) {
  return (
    <div className="video-grid-header">
      <span className="title">{props.title}</span>
    </div>
  );
}
