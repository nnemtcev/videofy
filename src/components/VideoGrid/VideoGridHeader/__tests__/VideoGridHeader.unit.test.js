import React from "react";
import { shallow } from "enzyme";
import { VideoGridHeader } from "../VideoGridHeader";

describe("VideoGridHeader", () => {
  test("renders with no props", () => {
    const wrapper = shallow(<VideoGridHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with title props set to empty string", () => {
    const wrapper = shallow(<VideoGridHeader title="" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with title props set to non-empty string", () => {
    const wrapper = shallow(<VideoGridHeader title="Trending" />);
    expect(wrapper).toMatchSnapshot();
  });
});
