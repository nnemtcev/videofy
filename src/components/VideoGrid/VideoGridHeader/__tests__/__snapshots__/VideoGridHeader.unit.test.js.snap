// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGridHeader renders with no props 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  />
</div>
`;

exports[`VideoGridHeader renders with title props set to empty string 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  />
</div>
`;

exports[`VideoGridHeader renders with title props set to non-empty string 1`] = `
<div
  className="video-grid-header"
>
  <span
    className="title"
  >
    Trending
  </span>
</div>
`;
